checker.c

```diff
@@ -1,21 +1,40 @@
 #include <stdio.h>
 #include <assert.h>
+#include "BatteryManagementConfig.h"
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+int alertIfBatteryIsUnHealthy(float MeasuredValue, BatteryManageConfig MinMaxMessage)
+{
+	if(MeasuredValue < MinMaxMessage.min_range) 
+	{
+    printf("%s\n", MinMaxMessage.Min_Alertmessage);
+	return HEALTH_NOT_OK;
+	}
+	else if(MeasuredValue > MinMaxMessage.max_range) 
+	{
+    printf("%s\n", MinMaxMessage.Max_Alertmessage);
+	return HEALTH_NOT_OK;
+	}
+    return HEALTH_OK;
+
+}
+
+int batteryIsOk(float temperature, float soc, float chargeRate) 
+{
+	return 	(alertIfBatteryIsUnHealthy(temperature, temperature_config) & \
```

Comment:
Complexity of this function keeps increasing with number of parameters. Any way to avoid that? Think how we speak about the parameter checks: 'battery is not ok if any of the parameters are not ok...'
What kind of code structure comes closest to that?